### PR TITLE
 Dedicated property on colDef to suppress grid export

### LINKED ISSUE #485: Provide dedicated property on colDef to suppress grid export
We have `ExportManager. getExportableColumns()` to filter the set of columns that are exported via Hoist's new server-side grid export. Currently, it returns all cols with a `field`: https://github.com/exhi/hoist-react/blob/4d683e7a97eacd34fd342a2aed8dc1204e7f5a0a/desktop/cmp/grid/ExportManager.js#L112

We need to be able to manually exclude other columns from an export, even if they do have a field set. These could include e.g. smaller metadata cols that only really make sense in the context of the interactive app and aren't desired in an exported file.

### PR DESCRIPTION
**Pull request by amcclain**
We have a use-case in a client app where we need to exclude certain cols from the export that do have fields but are just for use within the interactive app. This looked to me like the lightest-weight option for now to support such a thing.

@cnrudd or @TomTirapani if either of you have a chance to take a look tomorrow would appreciate. I was running into an issue on my Mac here where opening an exported file was crashing Excel - couldn't be related to this, some kind of file format / Excel version issue - but it means I wasn't able to fully test and would appreciate if you could do so quickly.

Fixes #485

### CODE REVIEW

File: columns/Utils.js [hunk 1]
```diff
@@ -63,15 +63,18 @@ export function fileColFactory(fileVals = {}) {
...
-            // Default exportName from headerName or field
+            // Default exportName from headerName or field.
             ret.exportName = ret.exportName || ret.headerName || startCase(ret.field);
 
-            // Use default exportFormat if not defined
+            // Use default exportFormat if not defined.
             ret.exportFormat = ret.exportFormat || ExportFormat.DEFAULT;
 
+            // Default excludeFromExport to true for all cols w/o a field.
+            ret.excludeFromExport = ret.excludeFromExport || !ret.field;
```
[the remark this answers is not in the document]
Comment (amcclain) [reply columns/Utils.js]: Agreed, if a field name were the number zero (or boolean false for that matter) that would be an an issue. Not sure if ag-Grid (or our data package) really intends to support such a value for field - I kind of doubt it - but I think it's always a good idea to point out possible falsely errors and a null check would be better.  

I would use `==` though as we would also skip if field were `undefined`. (Still think in general we can use `==` intentionally for these kinds of null/undef checks - another option would be to import lodash `isNil()` but that feels to me like overkill.

**Comment by lbwexler** [comment]
Agree about == but think best would be the falsey check actually. The check for existence of a real name is a great use of a Boolean test.  Missing the undefined vs. null vs. empty string is actually the real risk.

**Comment by amcclain** [comment]
Thanks to you both for looking over this - I am going to go ahead and merge it now - it's a small change and I expect this will be different very soon as I think we should have a proper class to describe these colDefs and their extensions (#416).

**Comment by amcclain** [comment]
Oh and of course we need to follow up on Excel crashing!  Added #499.

**Pull request merged**